src/rdb/rdb_raft.c

```diff
@@ -1682,6 +1765,12 @@ rdb_raft_save_state(struct rdb *db, struct rdb_raft_state *state)
 	state->drs_committed = raft_get_commit_idx(db->d_raft);
 }
 
+/* kccain - raft calls:
+ *  raft_is_leader()
+ *  raft_get_current_term()
+ *  raft_get_commit_idx()
+ *  raft_become_follower()
+ */
 /*
  * Check the current state against "state", which shall be a previously-saved
  * state, and handle any changes and errors.
```

Comment:
Right, rdb_raft_check_state shall be called with d_raft_mutex held. We shall release d_raft_mutex after this function returns (or this function shall release d_raft_mutex itself after the CV broadcast).